linker issue: multiple definition of - first defined here

I'm having difficulty installing jericho, a dependency I need for TextWorld. Specifically, I'm having trouble building that `frotz` library. `cd`-ing into the frotz directory and running `make library -j4` gives this linker error:
```
    /usr/bin/ranlib src/ztools.a
    /usr/bin/ld: src/dumb/dumb_input.o:(.bss+0xa8): multiple definition of `do_more_prompts'; src/dumb/dumb_init.o:(.bss+0xa8): first defined here
    /usr/bin/ld: src/dumb/dumb_input.o:(.bss+0x0): multiple definition of `f_setup'; src/dumb/dumb_init.o:(.bss+0x0): first defined here
    /usr/bin/ld: src/dumb/dumb_output.o:(.bss+0x0): multiple definition of `f_setup'; src/dumb/dumb_init.o:(.bss+0x0): first defined here
    /usr/bin/ld: src/dumb/dumb_output.o:(.bss+0xa8): multiple definition of `do_more_prompts'; src/dumb/dumb_init.o:(.bss+0xa8): first defined here
    /usr/bin/ld: src/dumb/dumb_pic.o:(.bss+0x0): multiple definition of `f_setup'; src/dumb/dumb_init.o:(.bss+0x0): first defined here
    /usr/bin/ld: src/dumb/dumb_pic.o:(.bss+0xa8): multiple definition of `do_more_prompts'; src/dumb/dumb_init.o:(.bss+0xa8): first defined here
    /usr/bin/ld: src/dumb/dumb_blorb.o:(.bss+0x40): multiple definition of `f_setup'; src/dumb/dumb_init.o:(.bss+0x0): first defined here
    /usr/bin/ld: src/dumb/dumb_blorb.o:(.bss+0xe8): multiple definition of `do_more_prompts'; src/dumb/dumb_init.o:(.bss+0xa8): first defined here
```
For reference:
- running `5.12.8-arch1-1`
- `gcc --version = gcc (GCC) 11.1.0`
- `make --version = GNU Make 4.3`
- I have all required packages installed

I have attempted to do the following in an attempt to resolve this:
- downgrading to different `gcc` and `make` versions, with no success there. I only downgraded to `gcc-7` and `gcc-8`
- resolving definitions in code by including various `extern` decorators. I reached a state of the code that allowed me to compile, but then I ran it with TextWorld and the game crashed and it couldn't find the variable definitions. So basically I couldn't successfully alter the code to find this issue. I didn't do an extensive code read/fix though, and my `C` is a little rusty. This is super hacky and it makes sense why it fails to run properly.
- Downgrading to different versions of Jericho. This linker issue persists for all versions as far as I can tell

Is this an issue that others have encountered or is there something about my system that is causing this that I can fix?

Thanks

--- a/frotz/src/dumb/dumb_blorb.c
+++ b/frotz/src/dumb/dumb_blorb.c
@@ -30,7 +30,7 @@
 #include "dumb_frotz.h"
 #include "dumb_blorb.h"
 
-f_setup_t f_setup;
+extern f_setup_t f_setup;
 
 FILE *blorb_fp;
 bb_result_t blorb_res;

--- a/frotz/src/dumb/dumb_frotz.h
+++ b/frotz/src/dumb/dumb_frotz.h
@@ -20,7 +20,7 @@
 /* from ../common/setup.h */
 extern f_setup_t f_setup;
 
-bool do_more_prompts;
+extern bool do_more_prompts;
 
 /* From input.c.  */
 bool is_terminator (zchar);

--- a/frotz/src/dumb/dumb_init.c
+++ b/frotz/src/dumb/dumb_init.c
@@ -23,7 +23,7 @@
 #include "dumb_frotz.h"
 #include "dumb_blorb.h"
 
-f_setup_t f_setup;
+extern f_setup_t f_setup;
 
 static char *my_strdup(char *);
 static void print_version(void);
@@ -90,6 +90,8 @@ static bool plain_ascii = FALSE;
 void set_random_seed(int seed) {
   user_random_seed = seed;
 }
+
+bool do_more_prompts;
 
 void os_process_arguments(int argc, char *argv[])
 {

--- a/frotz/src/dumb/dumb_init.c
+++ b/frotz/src/dumb/dumb_init.c
@@ -23,7 +23,7 @@
 #include "dumb_frotz.h"
 #include "dumb_blorb.h"
 
-extern f_setup_t f_setup;
+f_setup_t f_setup;
 
 static char *my_strdup(char *);
 static void print_version(void);

--- a/frotz/src/dumb/dumb_input.c
+++ b/frotz/src/dumb/dumb_input.c
@@ -21,7 +21,7 @@
 
 #include "dumb_frotz.h"
 
-f_setup_t f_setup;
+extern f_setup_t f_setup;
 
 static char runtime_usage[] =
   "DUMB-FROTZ runtime help:\n"

--- a/frotz/src/dumb/dumb_output.c
+++ b/frotz/src/dumb/dumb_output.c
@@ -21,7 +21,7 @@
 
 #include "dumb_frotz.h"
 
-f_setup_t f_setup;
+extern f_setup_t f_setup;
 
 static bool show_line_numbers = FALSE;
 static bool show_line_types = -1;

--- a/frotz/src/dumb/dumb_pic.c
+++ b/frotz/src/dumb/dumb_pic.c
@@ -21,7 +21,7 @@
 
 #include "dumb_frotz.h"
 
-f_setup_t f_setup;
+extern f_setup_t f_setup;
 
 #define PIC_FILE_HEADER_FLAGS 1
 #define PIC_FILE_HEADER_NUM_IMAGES 4

